src/GameData.ts

```diff
@@ -0,0 +1,265 @@
+            const item = returnData.filter(r => r.id === +i.id)[0];
+
+            i.from.forEach((f: number) => {
+                const otherItem = returnData.filter(r => r.id === f)[0];
+                if (otherItem !== undefined) {
+                    item.from.push(otherItem.name);
+                }
+            });
+
+            i.to.forEach((f: number) => {
+                const otherItem = returnData.filter(r => r.id === f)[0];
+                if (otherItem !== undefined) {
+                    item.to.push(otherItem.name);
+                }
+            });
+        });
+
+        return returnData;
+    }
+
+    public onLookup(message: Discord.Message, isAdmin: boolean, command: string, args: string[]) {
+        if (message.cleanContent.length === 0) {
+            let response = `I have info on the following categories; \`item\`,\`perk\`,\`champion\``;
+            response += `type !${command} {search_type} {search_term} to use it`;
+            message.channel.send(response);
+            return;
+        }
+
+        if (!["item", "perk", "champion"].some(i => i === args[0])) {
+            message.channel.send(`I'm sorry. I'm unable to parse the category \`${args[0]}\` at this moment. If you want it added, contact a guru`);
+            return;
+        }
+
+        if (args[0] === "item") {
+            message.channel.send(this.findItem(args.slice(1).join(" ")));
+            return;
+        }
+
+        if (args[0] === "perk") {
+            message.channel.send(this.findPerk(args.slice(1).join(" ")));
+            return;
+        }
+
+        if (args[0] === "champion") {
+            message.channel.send(this.findChampion(args.slice(1).join(" ")));
+            return;
+        }
+    }
+
+    public findItem(search: string): string {
+        if (!search) {
+            return `There are currently ${this.itemData.length} items in my lookup data!`;
+        }
+
+        const result = this.itemData.filter(c => Math.min(
+            ...[
+                levenshteinDistance(search, c.id.toString()),
+                levenshteinDistance(search, c.name),
+                levenshteinDistance(search, c.categories.toString()),
+            ]) < 1);
+
+        if (result.length === 0) {
+            return "Unable to find any data for that search term.";
+        }
+
+        if (result.length > 3) {
+            let response = "Too many results returned for that search, please try one of the options below:\n";
+            result.forEach(r => response += `\`${r.name}\`, `);
+            response = response.slice(0, -2);
+            return response;
+        }
+
+        const returnedValue = {
+            id: result[0].id,
+            name: result[0].name,
+            combineCost: result[0].price,
+            cost: result[0].priceTotal,
+            from: result[0].from,
+            to: result[0].to,
+        };
+
+        return "```" + JSON.stringify(returnedValue, null, 4) + "```";
```

Comment:
Why isn't this just `JSON.stringify(result[0], null, 4)`? Why are you creating a shallow clone?

Reply:
Because `result[0]` contains all the fields from the json data (even with the ItemData[] cast).
The resulting json has over 2000 chars for most (if not all) items, so discord cant send it